### ISSUE TITLE
Error using macros needed for decl_storage within substrate node template

### ISSUE THREAD

**Issue by shawntabrizi** [paritytech/substrate#1741]
```
error[E0463]: can't find crate for `hidden_include`
  --> /Users/shawntabrizi/Desktop/substrate-node-template/runtime/src/demo.rs:40:1
   |
40 | / decl_storage! {
41 | |     trait Store for Module<T: Trait> as Demo {
42 | |         Payment get(payment): Option<T::Balance>;
43 | |         Pot get(pot): T::Balance;
44 | |     }
45 | | }
   | |_^ can't find crate
```

This is the original error reported in https://github.com/paritytech/substrate/pull/1637#issuecomment-461598944

When changing the support macro from `support` to `srml_support` other erros appear:

```
error: cannot find macro `__dispatch_impl_metadata!` in this scope
  --> /Users/shawntabrizi/Desktop/substrate-node-template/runtime/src/demo.rs:11:1
   |
11 | / decl_module! {
12 | |     pub struct Module<T: Trait> for enum Call where origin: T::Origin {
13 | |         fn play(origin) -> Result {
14 | |             let sender = ensure_signed(origin)?;
...  |
37 | |     }
38 | | }
   | |_^
   |
```

**Comment by bkchr** [comment]
The first error can not be solved by us, we need to wait for rust to fix this.
This means, we can not rename crates in `Cargo.toml`. If you want to rename `srml_support`, do it in `lib.rs` with `extern crate srml_support as support;`.

For the second problem, you need to fix [this](https://github.com/paritytech/substrate/blob/master/srml/support/src/dispatch.rs#L735) to `$crate::__dispatch_impl_metadata! {`.

**Comment by shawntabrizi** [comment]
Solved with #1742
